contracts/src/v0.8/operatorforwarder/dev/AuthorizedForwarder.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ConfirmedOwnerWithProposal} from "../../shared/access/ConfirmedOwnerWithProposal.sol";
+import {AuthorizedReceiver} from "./AuthorizedReceiver.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+// solhint-disable custom-errors
+contract AuthorizedForwarder is ConfirmedOwnerWithProposal, AuthorizedReceiver {
+  using Address for address;
+
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  address public immutable getChainlinkToken;
+
+  event OwnershipTransferRequestedWithMessage(address indexed from, address indexed to, bytes message);
+
+  constructor(
+    address link,
+    address owner,
+    address recipient,
+    bytes memory message
+  ) ConfirmedOwnerWithProposal(owner, recipient) {
+    require(link != address(0), "Link token cannot be a zero address");
+    getChainlinkToken = link;
+    if (recipient != address(0)) {
+      emit OwnershipTransferRequestedWithMessage(owner, recipient, message);
+    }
+  }
+
+  /**
+   * @notice The type and version of this contract
+   * @return Type and version string
+   */
+  function typeAndVersion() external pure virtual returns (string memory) {
+    return "AuthorizedForwarder 1.0.0";
+  }
+
+  /**
+   * @notice Forward a call to another contract
+   * @dev Only callable by an authorized sender
+   * @param to address
+   * @param data to forward
+   */
+  function forward(address to, bytes calldata data) external validateAuthorizedSender {
+    require(to != getChainlinkToken, "Cannot forward to Link token");
+    _forward(to, data);
+  }
+
+  /**
+   * @notice Forward multiple calls to other contracts in a multicall style
+   * @dev Only callable by an authorized sender
+   * @param tos An array of addresses to forward the calls to
+   * @param datas An array of data to forward to each corresponding address
+   */
+  function multiForward(address[] calldata tos, bytes[] calldata datas) external validateAuthorizedSender {
+    require(tos.length == datas.length, "Arrays must have the same length");
+
+    for (uint256 i = 0; i < tos.length; i++) {
```

Comment:
always prefer ++i over i++ in loop increments, it's cheaper

Reply:
done